markdown-include.js

```diff
@@ -104,6 +105,41 @@ exports.buildLinkString = function (str) {
 	return str.trim().split(' ').join('-').toLowerCase();
 };
 
+exports.compileFolders = function (inDir, outDir) {
+	var deferred = q.defer();
+	var self = this;
+
+	fs.readdir(inDir, function (err, files) {
+		for (var i = 0; i < files.length; i++) {
+
+         var file = files[i];
+
+         // Skip directories
+			if (fs.statSync(file).isDirectory()) {
+				continue;
+         }
+         
+			var outFile = path.join(outDir, file);
+
+			self.processFile(file);
+			self.build[file].parsedData = self.stripTagsInFile({
+				data: self.build[file].parsedData,
+				pattern: self.ignorePattern,
+				string: self.ignoreTag
+			});
+
+			if (self.customTags && self.customTags.length) {
+				self.build[file].parsedData = self.resolveCustomTags(self.build[file].parsedData);
+			}
+
+			self.writeFile(self.build[file].parsedData, outFile);
```

Comment:
The use of deferred.resolve() here could potentially be called prior to the file completing the write, since your callback parameter doesn't actually include the deferred.resolve() making the return from compileFolders not exactly reliable

Reply:
Ah I see the issue. Would you recommend using async to keep track of all the file writes before calling resolve?